File: Python/0201-0250/206. Reverse Linked List.py
# https://leetcode.com/problems/reverse-linked-list/description/

class Solution:
    def reverseList(self, head: Optional[ListNode])  -> Optional[ListNode]:
        if head == None or head.next == None :
            return head # if one or no nodes in list
        
        # using 3 pointers
        previous = None
        current = head
        nexts = current.next

        while current != None :
            current.next = previous #  point current to backwards
            previous = current # move previous one step
            current = nexts # move current one step
            if nexts!=None :
                nexts = nexts.next
            
        return previous

File: Javascript/0201-0250/206. Reverse Linked List.js
// https://leetcode.com/problems/reverse-linked-list/description/

var reverseList = function(head) {
    if(head == null || head.next ==null){
        return head; //if one or no nodes in list
    }
    //using 3 pointers
    previous = null;
    current = head;
    next = current.next;

    while(current != null){
        current.next = previous; // point current to backwards
        previous = current; //move previous one step
        current = next; //move current one step
        if(next!=null){
            next = next.next;
        }
    }

    return previous;
};
